project/Elasticsearch.scala

```diff
@@ -0,0 +1,84 @@
+import sbt._
+import sbt.Keys._
+import sbt.dsl.LinterLevel.Ignore
+
+import scala.annotation.nowarn
+
+import org.testcontainers.containers.BindMode
+import org.testcontainers.containers.FixedHostPortGenericContainer
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy
+import org.testcontainers.dockerclient.DockerClientProviderStrategy
+
+import java.util.concurrent.TimeoutException
+import java.lang.Thread
+import java.net.HttpURLConnection
+import java.net.HttpURLConnection._
+import java.net.URL
+import java.time.Duration
+import java.io.IOException
+
+object Elasticsearch extends AutoPlugin {
+  object autoImport {
+    val startElasticsearch = taskKey[Unit](
+      "Check that elasticsearch has already started or else start a container"
+    )
+  }
+
+  import autoImport._
+
+  override lazy val globalSettings: Seq[Setting[_]] = Seq(
+    startElasticsearch := {
+      val esData = (ThisBuild / baseDirectory).value / ".esdata"
+      val logger = streams.value.log
+      if (alreadyStarted) {
+        logger.info("elasticsearch has already started")
+      } else {
+        logger.info("Trying to start elasticsearch container")
+        if (!esData.exists) IO.createDirectory(esData)
+        IO.setPermissions(esData, "rwxrwxrwx")
```

Comment:
Should we use more strict permissions?

Reply:
elasticsearch needs read and write access to it, but I don't know if I can change the container user

Reply:
It also needs execution right